zeppelin-web/src/app/interpreter/interpreter.html

```diff
@@ -41,12 +41,13 @@ <h3 class="new_h3" style="float:left">
       <div class="col-md-12">
         <h3 class="interpreter-title">{{setting.name}}
           <small>
-            <span ng-repeat="interpreter in setting.interpreterGroup"
+            <span style="display:inline-block" ng-repeat="interpreter in setting.interpreterGroup"
                   title="{{interpreter.class}}">
-              <span ng-show="$parent.$first && $first">%{{setting.name}} (default)</span>
-              <span ng-show="$parent.$first && !$first">, %{{interpreter.name}}</span>
-              <span ng-show="!$parent.$first && $first">%{{setting.name}}</span>
-              <span ng-show="!$parent.$first && !$first">, %{{setting.name}}.{{interpreter.name}}</span>
+              <span ng-show="!$first">, </span>
+              %<span ng-show="!$parent.$first || $first">{{setting.group}}</span
+              ><span ng-show="(!$parent.$first || $first) && !$first">.</span
```

Comment:
Similarly here. To elaborate, if we want to show `.` for all interpreters not the first in the group, then it should be if `!$first`